### PR TITLE
Add CI mode to training

### PR DESCRIPTION
**Pull request by Parzival-05**

### CODE REVIEW

**Comment by gsvgit** [comment]
@Parzival-05 Please, rebase it on current `main`.

File: AIAgent/ml/training/validation.py [hunk 1]
```diff
@@ -82,11 +84,16 @@ def validate_coverage(
             if isinstance(result, GameError):
                 need_to_save_map: bool = result.need_to_save_map()
                 if not need_to_save_map:
-                    statistics_collector.fail(result._map)
+                    if not fail_immediately:
+                        statistics_collector.fail(result._map)
+                    else:
+                        failed_validation = True
+                        break
```
Comment (gsvgit) [inline AIAgent/ml/training/validation.py:91]: We should try run all maps, but fail at the end if at least one failed.

File: AIAgent/ml/training/validation.py [hunk 2]
```diff
@@ -82,11 +84,16 @@ def validate_coverage(
             if isinstance(result, GameError):
                 need_to_save_map: bool = result.need_to_save_map()
                 if not need_to_save_map:
-                    statistics_collector.fail(result._map)
+                    if not fail_immediately:
+                        statistics_collector.fail(result._map)
```
[the remark this answers is not in the document]
Comment (Parzival-05) [reply AIAgent/ml/training/validation.py:88]: Logging is redundant here. The error information is already logged in "play_game". https://github.com/PySymGym/PySymGym/blob/9153be0dcb13b5241add2623e04c0443e245493a/AIAgent/ml/game/play_game.py#L186

File: AIAgent/ml/training/validation.py (hunk 1 above)
Comment (Parzival-05) [reply AIAgent/ml/training/validation.py]: https://github.com/PySymGym/PySymGym/pull/104/commits/e8b533cc9b54468fad767cee26916a9758829a4d

File: AIAgent/ml/training/validation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Parzival-05) [reply AIAgent/ml/training/validation.py:88]: https://github.com/PySymGym/PySymGym/pull/108

**Comment by Parzival-05** [comment]
@gsvgit https://github.com/PySymGym/PySymGym/pull/104/commits/9344d9fbd7a3aa5ea4a8395f0cadb9b4c787227e

**Pull request merged**